File: Modules/module_sample/src/main/java/com/example/william/my/module/sample/activity/MvpActivity.java
package com.example.william.my.module.sample.activity;

import android.os.Bundle;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.blankj.utilcode.util.FragmentUtils;
import com.example.william.my.library.base.BaseActivity;
import com.example.william.my.module.router.ARouterPath;
import com.example.william.my.module.sample.R;
import com.example.william.my.module.sample.fragment.MvpFragment;
import com.example.william.my.module.sample.presenter.ArticlePresenter;
import com.example.william.my.module.sample.repo.ArticleRepository;

@Route(path = ARouterPath.Sample.Sample_MVP)
public class MvpActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.sample_layout_fragment);

        MvpFragment mvpFragment = (MvpFragment) getSupportFragmentManager()
                .findFragmentById(R.id.contentFrame);

        if (mvpFragment == null) {
            mvpFragment = MvpFragment.newInstance();
            FragmentUtils.add(getSupportFragmentManager(), mvpFragment, R.id.contentFrame);
        }

        // Create the presenter
        new ArticlePresenter(ArticleRepository.getInstance(), mvpFragment);
    }
}

File: modules/module_sample/src/main/kotlin/com/example/william/my/module/sample/activity/MvpActivity.kt
package com.example.william.my.module.sample.activity

import androidx.fragment.app.Fragment
import com.alibaba.android.arouter.facade.annotation.Route
import com.example.william.my.library.base.BaseFragmentActivity
import com.example.william.my.module.router.ARouterPath
import com.example.william.my.module.sample.fragment.MvpFragment

@Route(path = ARouterPath.Sample.Sample_MVP)
class MvpActivity : BaseFragmentActivity() {

    override fun setFragment(): Fragment {
        return MvpFragment()
    }
}